src/library/Cardano/CoinSelection/Fee.hs

```diff
@@ -100,32 +105,21 @@ import qualified Data.List.NonEmpty as NE
 -- This type is isomorphic to 'Coin'.
 --
 newtype Fee = Fee
-    { unFee :: Word64 }
+    { unFee :: Integer }
```

Comment:
`Fee` can be negative? :thinking:

Reply:
> `Fee` can be negative?

In order to limit the scope of this PR, I have not (yet) attempted to restrict `Fee` in the same way as `Coin` and `DustThreshold`.

If we can determine that our codebase never creates any intermediate negative values of `Fee`, then we could also redefine `Fee` in terms of `SafeNatural`.

Reply:
@paweljakubas Fixed in 79fdc177d677ad1974ef426b836a0e644fd4a25e.